tests/e2e/specs/stories-editor/context-menu.js

```diff
@@ -0,0 +1,177 @@
+			const block = await page.$( BLOCK_SELECTOR );
+			await makeRightClick( block );
+
+			await clickButton( 'Copy Block' );
+
+			await removeAllBlocks();
+			await page.waitForSelector( ACTIVE_PAGE_SELECTOR );
+			const pageBlock = await page.$( ACTIVE_PAGE_SELECTOR );
+			await makeRightClick( pageBlock );
+
+			await clickButton( 'Paste' );
+			await page.waitForSelector( ACTIVE_PAGE_SELECTOR + ' ' + BLOCK_SELECTOR );
+			expect( page ).toMatchElement( ACTIVE_PAGE_SELECTOR + ' ' + BLOCK_SELECTOR );
+		} );
+
+		it( 'should allow cutting and pasting a block', async () => {
+			const block = await page.$( BLOCK_SELECTOR );
+			await makeRightClick( block );
+
+			await clickButton( 'Cut Block' );
+			expect( page ).not.toMatchElement( BLOCK_SELECTOR );
+
+			const pageBlock = await page.$( ACTIVE_PAGE_SELECTOR );
+			await makeRightClick( pageBlock );
+
+			await clickButton( 'Paste' );
+
+			await page.waitForSelector( BLOCK_SELECTOR );
+			expect( page ).toMatchElement( BLOCK_SELECTOR );
+		} );
+
+		it( 'should allow duplicating a block', async () => {
+			const block = await page.$( BLOCK_SELECTOR );
+			await makeRightClick( block );
+
+			await clickButton( 'Duplicate Block' );
+
+			const nodes = await page.$x(
+				'//div[contains(@class, "wp-block-amp-amp-story-post-author")]'
+			);
+			expect( nodes ).toHaveLength( 2 );
+		} );
+
+		it( 'should allow removing a block', async () => {
+			const block = await page.$( BLOCK_SELECTOR );
+			await makeRightClick( block );
+
+			await clickButton( 'Remove Block' );
+			expect( page ).not.toMatchElement( BLOCK_SELECTOR );
+		} );
+
+		it( 'should not allow pasting disallowed blocks', async () => {
+			const firstPageClientId = ( await getAllBlocks() )[ 0 ].clientId;
+			await insertBlock( 'Page' );
+			await insertBlock( 'Call to Action' );
+
+			const callToActionSelector = '.wp-block-amp-amp-story-cta';
+			const ctaBlock = await page.waitForSelector( callToActionSelector );
+			await makeRightClick( ctaBlock );
+
+			await clickButton( 'Copy Block' );
+
+			await goToPreviousPage();
+			await selectBlockByClientId( firstPageClientId );
+			const pageBlock = await page.$( `#block-${ firstPageClientId }` );
+			// Wait for transition time 300ms.
+			await page.waitFor( 300 );
+			await makeRightClick( pageBlock );
+
+			await clickButton( 'Paste' );
+			expect( page ).not.toMatchElement( `#block-${ firstPageClientId } ${ callToActionSelector }` );
+		} );
+
+		it( 'should not allow duplicate disallowed blocks', async () => {
+			await insertBlock( 'Page' );
+			await insertBlock( 'Page Attachment' );
+			const callToActionSelector = '.wp-block[data-type="amp/amp-story-page-attachment"]';
+			const ctaBlock = await page.waitForSelector( callToActionSelector );
+			await makeRightClick( ctaBlock );
+			const duplicateSelector = 'right-click-duplicate';
+			expect( page ).not.toMatchElement( duplicateSelector );
+		} );
+	} );
+
+	describe( 'using Keyboard', () => {
```

Comment:
Should we also add some other tests to check if the first element is selected and so users can navigate elements in our context menu using the down / up arrows.

Reply:
Again I feel a strong need to point out, that e2e's aren't meant for exhaustive covering-all-the-bases-testing - not by a long stretch. It's meant for a testing a few happy paths. Please bear in mind, that every test added also adds 5 seconds to the build time of **every single future PR**. I don't think it's worth it.

Testing that the `ContextMenu` works, should be done in a test for the `ContextMenu`. This is merely a test that the ContextMenu shows up when the correct keyboard combination is pressed.